B9PartSwitch/PartSwitch/PartModuleSwitchInfo.cs

```diff
@@ -48,48 +48,64 @@ public List<PartModule> FindModule(Part part)
                     // we are asked to find all modules of this type
                     if (findAll)
                     {
-                        modules.Add(part.Modules[i]);
-                        continue;
+                        yield return part.Modules[i];
                     }
 
                     // else we try to get the correct module
                     // try to use the field identifier
                     if (fieldIdentifier.Length != 0)
                     {
                         // get identifier
-                        string id = "";
                         BindingFlags flags = BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.FlattenHierarchy;
                         FieldInfo field = part.Modules[i].GetType().GetField(fieldIdentifier, flags);
-                        field.SetValue(part.Modules[i], id); // TODO: check what happens with invalid field name, maybe this should be in a try-catch ?
 
-                        // if the identifier value match
-                        if (id == valueIdentifier)
+                        if (field.FieldType == typeof(string))
```

Comment:
Actually, now that I think about it, B9PartSwitch already maintains a pretty extensive list of parsers in `B9PartSwitch.Fishbones.Parsers.DefaultValueParseMap` - it might be a good idea to use that rather than try and implement everything here.  I think it would look something like this:

```
IValueParseMap valueParseMap = new DefaultValueParseMap();
if (valueParseMap.CanParse(field.FieldType)
{
    try
    {
        if (object.Equals(valueParseMap.GetParser(field.FieldType).Parse(valueIdentifier), field.GetValue(part.Modules[i]))
        {
            yield return part.Modules[i];
        }
    }
    catch (Exception ex)
    {
        Debug.LogException(ex);
    }
}
else
{
   Debug.LogError($"Could not find a suitable way to parse type {field.FieldType}");
}
```